tests/testimageinterfaces.py

```diff
@@ -413,6 +413,22 @@ class ImageXArrayInterfaceTest(ImageGridInterfaceTest):
 
     datatype = 'xarray'
 
+    def setUp(self):
+        try:
+            import xarray as xr
+        except:
+            raise SkipTest('Test requires xarray')
+        super(ImageXArrayInterfaceTest, self).setUp()
+
+    def test_select_on_transposed_dataarray(self):
+        import xarray as xr
+        x = np.linspace(-3, 7, 533)
+        y = np.linspace(-5, 8, 894)
+        z = np.exp(-1*(x**2 + y[:, np.newaxis]**2))
+        array = xr.DataArray(z, coords=[y, x], dims=['x', 'y'])
+        img = Image(array)[1:3]
+        self.assertEqual(img.data, Image(array.sel(x=slice(1, 3))).data)
+
```

Comment:
Might be nice to have one more unit test that relies on the updated ``shape`` property implementation.